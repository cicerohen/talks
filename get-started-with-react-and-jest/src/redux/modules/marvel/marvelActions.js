import MODULE_NAME from './marvelConstants';

export const FETCH_CHARACTERS_REQUESTED = `${MODULE_NAME}/FETCH_CHARACTERS_REQUESTED`;
export const FETCH_CHARACTERS_COMPLETED = `${MODULE_NAME}/FETCH_CHARACTERS_COMPLETED`;
export const FETCH_CHARACTERS_FAILED = `${MODULE_NAME}/FETCH_CHARACTERS_FAILED`;

export const fetchCharacters = () => ({
  type: FETCH_CHARACTERS_REQUESTED,
});

export const completeFetchCharacters = data => ({
  type: FETCH_CHARACTERS_COMPLETED,
  payload: {
    data,
  },
});

export const failFetchCharacters = error => ({
  type: FETCH_CHARACTERS_FAILED,
  payload: {
    error,
  },
});
