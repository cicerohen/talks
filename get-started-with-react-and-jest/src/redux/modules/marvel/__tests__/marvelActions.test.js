import * as actions from '../marvelActions';

describe('marvel/marvelActions.js Tests', () => {
  it('should create fetchCharacters action object correctly', (done) => {
    expect(actions.fetchCharacters()).toMatchSnapshot();
    done();
  });

  it('should create completeFetchCharacters action object correctly', (done) => {
    const data = [{ id: 1, prop: 'prop' }];
    expect(actions.fetchCharacters(data)).toMatchSnapshot();
    done();
  });

  it('should create completeFetchCharacters action object correctly', (done) => {
    const error = { type: 'Error'};
    expect(actions.failFetchCharacters(error)).toMatchSnapshot();
    done();
  });
});