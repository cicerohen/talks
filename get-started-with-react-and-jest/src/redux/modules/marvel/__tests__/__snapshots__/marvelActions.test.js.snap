// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marvel/marvelActions.js Tests should create completeFetchCharacters action object correctly 1`] = `
Object {
  "type": "marvel/FETCH_CHARACTERS_REQUESTED",
}
`;

exports[`marvel/marvelActions.js Tests should create completeFetchCharacters action object correctly 2`] = `
Object {
  "payload": Object {
    "error": Object {
      "type": "Error",
    },
  },
  "type": "marvel/FETCH_CHARACTERS_FAILED",
}
`;

exports[`marvel/marvelActions.js Tests should create fetchCharacters action object correctly 1`] = `
Object {
  "type": "marvel/FETCH_CHARACTERS_REQUESTED",
}
`;
